src/core/Akka.Streams/Dsl/RestartFlow.cs

```diff
@@ -481,31 +187,42 @@ internal sealed class RestartWithBackoffFlow<TIn, TOut, TMat> : GraphStage<FlowS
 
         public override FlowShape<TIn, TOut> Shape { get; }
 
-        protected override GraphStageLogic CreateLogic(Attributes inheritedAttributes) => new Logic(this, "Flow");
+        protected override GraphStageLogic CreateLogic(Attributes inheritedAttributes) => new Logic(this, inheritedAttributes, "Flow");
 
         private sealed class Logic : RestartWithBackoffLogic<FlowShape<TIn, TOut>, TIn, TOut>
         {
             private readonly RestartWithBackoffFlow<TIn, TOut, TMat> _stage;
+            private readonly Attributes _inheritedAttributes;
             private Tuple<SubSourceOutlet<TIn>, SubSinkInlet<TOut>> _activeOutIn;
-
-            public Logic(RestartWithBackoffFlow<TIn, TOut, TMat> stage, string name)
+            private TimeSpan _delay;
+            
+            public Logic(RestartWithBackoffFlow<TIn, TOut, TMat> stage, Attributes inheritedAttributes, string name)
                 : base(name, stage.Shape, stage.In, stage.Out, stage.Settings, stage.OnlyOnFailures)
             {
+                _delay = _inheritedAttributes.GetAttribute<Delay>(new Delay(TimeSpan.FromMilliseconds(50))).Duration;
```

Comment:
Thats one thing im unsure about. 
In the scala source, the `Delay` class is a case class, wrapped in an object. (see: https://github.com/akka/akka/blob/master/akka-stream/src/main/scala/akka/stream/scaladsl/RestartFlow.scala#L402)  Im unsure if that means its a public api or not. But i figured undefined access modifiers means private/internal so i went with that.

The comments on the original scala bugfix suggest its overridable through a WithAttributes call on the RestartFlow, but im not sure if that works here (regardless of the access modifier). im pretty sure they are not applied to the flow the RestartFlow is wrapping.

However i think normally nobody will ever want to tweak stuff in their stream on this level. Im of the opinion that "nobody should ever need to tweak this, because it should just work". But considering what the fix is, they might have to ? Although i'd expect that the # of ppl that would need to do this would be vanishingly rare.

Reply:
Yeah that's a public API - it's meant to be an attribute you can pass in at the time the flow is declared via the `inheritedAttributes` collection. So I guess, broadly speaking, this is meant to be configurable via the attributes approach.

Reply:
> Although i'd expect that the # of ppl that would need to do this would be vanishingly rare.

Agree - so the duration here just sets a 50ms threshold for restarting. That seems fine for most cases, where it's recovering from a crash.